Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm

```diff
@@ -768,10 +768,19 @@
         return { url, DidFilterLinkDecoration::No };
 
     auto sanitizedURL = url;
+    auto removedParameters = WTF::removeQueryParameters(sanitizedURL, [&](const auto& parameter) {
+        auto it = m_linkDecorationFilteringData.find(parameter);
+        if (it == m_linkDecorationFilteringData.end())
+            return false;
+
+        const auto& conditionals = it->value;
+        bool missingOrFoundDomain = conditionals.domains.isEmpty() || conditionals.domains.contains(RegistrableDomain { url });
+        bool missingOrFoundPath = conditionals.paths.isEmpty() || std::any_of(conditionals.paths.begin(), conditionals.paths.end(),
```

Comment:
Nit: something like `unconditional`/`unrestricted`/`empty` might be clearer (e.g., `unconditionalOrFoundDomain`). To me `missing` implies that something is potentially wrong.

Reply:
👍 I’ll change this to `emptyOrFoundDomain`.

Reply:
That's not a great name easy. Per coding style, boolean variables in WebKit need to have a prefix such as 'is' / 'has'.

Reply:
Ok, I'll use `isEmptyOrFoundDomain`.